vortex-array/src/compute/take.rs

```diff
@@ -112,7 +113,7 @@ fn take_impl(array: &Array, indices: &Array, checked_indices: bool) -> VortexRes
         } else {
             take_fn.take(array, indices)
         }?;
-        if array.dtype() != result.dtype() {
+        if !array.dtype().eq_ignore_nullability(result.dtype()) {
```

Comment:
i know this isn't new, but this check should probably just be a debug_assert

Reply:
Yes, there were two of them. I was debating if this is just debug or regular but I guess the pattern is to have debug asserts